drivers/counter/counter_mcux_lptmr.c

```diff
@@ -198,59 +198,66 @@ static const struct counter_driver_api mcux_lptmr_driver_api = {
-#if DT_INST_PROP(0, prescaler) == 1
-	.bypass_prescaler_glitch = true,
-#else
-	.prescaler_glitch = TO_LPTMR_GLITCH(DT_INST_PROP(0, prescaler)),
-#endif
-	.mode = kLPTMR_TimerModePulseCounter,
-	.pin = TO_LPTMR_PIN_SEL(DT_INST_PROP(0, input_pin)),
-	.polarity = DT_INST_PROP(0, active_low),
-#else /* !DT_INST_NODE_HAS_PROP(0, input_pin) */
-	.mode = kLPTMR_TimerModeTimeCounter,
-#if DT_INST_PROP(0, prescaler) == 1
-	.bypass_prescaler_glitch = true,
-#else
-	.prescaler_glitch = TO_LPTMR_PRESCALER(DT_INST_PROP(0, prescaler)),
-#endif
-#endif /* !DT_INST_NODE_HAS_PROP(0, input_pin) */
-	.irq_config_func = mcux_lptmr_irq_config_0,
-};
 
-DEVICE_DT_INST_DEFINE(0, &mcux_lptmr_init, NULL,
-		    &mcux_lptmr_data_0,
-		    &mcux_lptmr_config_0,
-		    POST_KERNEL, CONFIG_COUNTER_INIT_PRIORITY,
-		    &mcux_lptmr_driver_api);
+#define COUNTER_MCUX_SET_INPUT_PIN_CONFIG_VALUES(n)				\
+	.mode = COND_CODE_1(DT_INST_NODE_HAS_PROP(n, input_pin),		\
+			(kLPTMR_TimerModePulseCounter),				\
+			(kLPTMR_TimerModeTimeCounter)),				\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(n, input_pin),			\
+		(.pin = TO_LPTMR_PIN_SEL(DT_INST_PROP(n, input_pin)),),		\
+		())								\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(n, input_pin),			\
+		(.polarity = DT_INST_PROP(n, active_low),),			\
+		())
 
-static void mcux_lptmr_irq_config_0(const struct device *dev)
-{
-	IRQ_CONNECT(DT_INST_IRQN(0), DT_INST_IRQ(0, priority),
-		    mcux_lptmr_isr, DEVICE_DT_INST_GET(0), 0);
-	irq_enable(DT_INST_IRQN(0));
-}
-#endif	/* DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay) */
+/*
+ * If prescaler value is set to 1, we treat it as a division of 1 otherwise
+ * we set it to one of the enums above
+ */
+#define COUNTER_MCUX_SET_PRESCALER_GLITCH_VALUE(n)				\
+	COND_CODE_1(DT_INST_PROP(n, prescaler),					\
+			(1),							\
+			(COND_CODE_1(DT_INST_PROP(				\
+				n, counter_mode_enable),			\
+			(.prescaler_glitch = TO_LPTMR_PRESCALER(		\
+				DT_INST_PROP(n, prescaler))),			\
+			(.prescaler_glitch = TO_LPTMR_GLITCH(			\
+				DT_INST_PROP(n, prescaler)))))),
+
+
+#define COUNTER_MCUX_LPTMR_DEVICE_INIT(n)					\
+	static void mcux_lptmr_irq_config_##n(const struct device *dev)		\
+	{									\
+		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority),		\
+			mcux_lptmr_isr, DEVICE_DT_INST_GET(n), 0);		\
+		irq_enable(DT_INST_IRQN(n));					\
+	}									\
+										\
+	static struct mcux_lptmr_data mcux_lptmr_data_##n;			\
+	static void mcux_lptmr_irq_config_##n(const struct device *dev);	\
+	static struct mcux_lptmr_config mcux_lptmr_config_##n = {		\
+		.info = {							\
+			.max_top_value = ((DT_INST_PROP(n, resolution) == 32)	\
+				? UINT32_MAX : UINT16_MAX),			\
+			.freq = DT_INST_PROP(n, clock_frequency) /		\
+				DT_INST_PROP(n, prescaler),			\
+			.flags = COUNTER_CONFIG_INFO_COUNT_UP,			\
+			.channels = 0,						\
+		},								\
+		.base = (LPTMR_Type *)DT_INST_REG_ADDR(n),			\
+		.clk_source = TO_LPTMR_CLK_SEL(DT_INST_PROP(n, clk_source)),	\
+		.bypass_prescaler_glitch = (bool)DT_INST_PROP(n,		\
```

Comment:
doesn't need to be casted, also, why is this being done, and has it been tested? I don't see any actual code change above, what will this actually do?

Reply:
This has been tested.